bluepyentity/entity_definitions.py

```diff
@@ -0,0 +1,230 @@
+"""WIP: pydantic datamodels with some of the functionality"""
+import datetime
+import json
+import pathlib
+from typing import List, Union
+
+import pydantic
+
+# Custom pydantic types
+
+
+class StrToList:
+    """Helper class to force strings to list of strings."""
+
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.str_to_list
+
+    @staticmethod
+    def str_to_list(item):
+        if isinstance(item, str):
+            return [item]
+
+        raise TypeError("str type expected")
+
+
+class DataDownload:
+    """Helper class to convert path to DataDownload like structure."""
+
+    # NOTE: functionality should likely be in the adapter class / registration functions, in which
+    # we could handle also other downloadable data creation like in entity-management. This could
+    # become a mere subclass of it for config files.
+
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.wrap
+
+    @staticmethod
+    def wrap(path):
+        from bluepyentity.register import _wrap_linked_file_path
+
+        if isinstance(path, str):
+            return {"type": "DataDownload", "url": _wrap_linked_file_path(path)}
+
+        raise TypeError("str type expected")
+
+
+ListOfStr = Union[StrToList, List[str]]
+
+
+# Data Models
+
+
+class BaseModel(pydantic.BaseModel):
```

Comment:
I haven't used pydantic before, and I am curious what would be the advantage over using the builtin dataclasses instead?